pandas/core/generic.py

```diff
@@ -6362,6 +6362,17 @@ def fillna(
         1   3.0 4.0 NaN 1
         2   NaN 1.0 NaN 5
         3   NaN 3.0 NaN 4
+
+        When filling using a DataFrame, replacement happens along
+        the same column names and same indices
+
+        >>> df2 = pd.DataFrame(np.zeros((5, 5)), columns=list("ABCED"))
```

Comment:
could `df2` have columns `"ABCE"`, so it's clear that there will be no replacement in column D (which is absent from `df2`)?

Reply:
Changed